### ISSUE TITLE
Can't subscribe to `YepAdvancementEvent`

### ISSUE THREAD

**Issue by axd1x8a** [unilock/YepLib#1]
I'm trying to listen for `YepAdvancementEvent` but getting this error in console:
```
java.lang.ClassCastException: Cannot cast cc.unilock.yeplib.api.event.YepMessageEvent to cc.unilock.yeplib.api.event.YepAdvancementEvent
```
example of code:
```java
@Subscribe
public void onYepAdvancement(YepAdvancementEvent event) {
    logger.info("Handling YepAdvancementEvent: {}", event.toString());
}

@Subscribe
public void onYepDeath(YepDeathEvent event) {
    logger.info("Handling YepDeathEvent: {}", event.toString());
}
```

**Comment by unilock** [comment]
That's strange; that error should have been fixed as of https://github.com/unilock/YepLib/commit/ee3ebe3234ad79650cb22a553bcd79a55d90c299 (v2.1.0 and above).

I'm honestly not sure what the issue could be outside of making sure you're using the latest version of YepLib. (I asked about it in PaperMC's support Discord; even they have no idea what's going on...)

**Comment by axd1x8a** [comment]
Looked in the logs, everything is working, it's just that for some reason velocity is sending all yep events to handlers that expect specific ones

**Comment by unilock** [comment]
This should be fixed with 2.3.0, god willing
